packages/mongo/mongo_driver.js

```diff
@@ -1043,6 +1043,20 @@ class AsynchronousCursor {
   [Symbol.iterator]() {
     return this._cursor[Symbol.iterator]();
   }
+  
+  [Symbol.asyncIterator]() {
+    return {
+      next: async () => {
+        const next = await this._nextObjectPromise();
+        if (next) {
+          return { done: false, value: next };
+        }
+        return {
+          done: true,
+        };
+      },
```

Comment:
This can be simplified:
```suggestion
      async next {
        const value = await this._nextObjectPromise();
        return { done: !value, value };
      },
```